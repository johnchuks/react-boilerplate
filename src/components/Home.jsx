import React from 'react';
import { Link } from 'react-router-dom';

const Home = () => (
  <div>
    <h1>Welcome to my React Boilerplate</h1>&nbsp;

    <Link to='/sample'>SampleLink</Link>
    </div>
)
export default Home;
