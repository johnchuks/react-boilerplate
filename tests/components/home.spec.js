import React from 'react';
import Home from '../../src/components/Home.jsx';

test('renders the welcome message', () => {
  const wrapper = shallow(<Home />);
  expect(wrapper).toMatchSnapshot();
});
