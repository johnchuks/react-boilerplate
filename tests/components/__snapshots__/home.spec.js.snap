// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the welcome message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Welcome to my React Boilerplate
        </h1>,
        " ",
        <Link
          replace={false}
          to="/about"
        >
          About
        </Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to my React Boilerplate",
        },
        "ref": null,
        "rendered": "Welcome to my React Boilerplate",
        "type": "h1",
      },
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "About",
          "replace": false,
          "to": "/about",
        },
        "ref": null,
        "rendered": "About",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Welcome to my React Boilerplate
          </h1>,
          " ",
          <Link
            replace={false}
            to="/about"
          >
            About
          </Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to my React Boilerplate",
          },
          "ref": null,
          "rendered": "Welcome to my React Boilerplate",
          "type": "h1",
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "About",
            "replace": false,
            "to": "/about",
          },
          "ref": null,
          "rendered": "About",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
